packages/provider/src/multi-wallet.ts

```diff
@@ -0,0 +1,271 @@
+import { Signer as AbstractSigner, Contract, ethers } from 'ethers'
+import { TransactionResponse, JsonRpcProvider } from 'ethers/providers'
+import { Wallet } from './wallet'
+import { NetworkConfig, ArcadeumWalletConfig, ArcadeumContext, Transactionish } from './types'
+import { Arrayish, BigNumberish } from 'ethers/utils'
+import { abi as mainModuleUpgradableAbi } from './abi/mainModuleUpgradable'
+import { abi as requireUtilsAbi } from './abi/requireUtils'
+import { isConfig } from './utils'
+import { NotEnoughSigners } from './errors'
+
+// TODO: Add more details to network
+// authChain, mainNetwork, etc
+export type NetworkWallet = {
```

Comment:
btw, what is a "network wallet" ?

Reply:
It's a wallet for a given network, each network has it's own "internal" wallet